src/components/Header.tsx

```diff
@@ -0,0 +1,166 @@
+import React, { useState, useRef } from "react";
+import { Modal } from "antd";
+
+import { FaRegBell, FaCaretDown } from "react-icons/fa";
+import { PiSignOut, PiGearSix } from "react-icons/pi";
+import { FaCircleUser } from "react-icons/fa6";
+import { AiOutlineUser } from "react-icons/ai";
+import { motion, AnimatePresence } from "framer-motion";
+
+import { useNavigate } from "react-router-dom";
+
+import { MdOutlineMenu } from "react-icons/md";
+
+interface UserHedeprops {
+  active: boolean;
+  setActive: (active: boolean) => void;
+}
+
+const Header: React.FC<UserHedeprops> = ({ active, setActive }) => {
+  const [showMenu, setShowMenu] = useState(false);
+  const [userImage, setUserImage] = useState<string | null>(null);
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const navigate = useNavigate();
+  const fileInputRef = useRef<HTMLInputElement>(null);
+
+  const handleShowMenu = () => {
+    setShowMenu(!showMenu);
+  };
+
+  const handleUserIconClick = () => {
+    setIsModalVisible(true);
+  };
+
+  const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.files && event.target.files[0]) {
+      const file = event.target.files[0];
+      const reader = new FileReader();
+      reader.onloadend = () => {
+        setUserImage(reader.result as string);
+      };
+      reader.readAsDataURL(file);
+    }
+  };
+
+  const handleOk = () => {
+    setIsModalVisible(false);
+    if (fileInputRef.current) {
+      fileInputRef.current.click();
+    }
+  };
+
+  const handleCancel = () => {
+    setIsModalVisible(false);
+  };
+
+  const handleMenubar = () => {
+    setActive(!active);
+    // console.log(active);
+  };
+
+  const menuItems = [
+    {
+      label: "My Profile",
+      icon: <AiOutlineUser className="text-2xl" />,
+      path: "account/profile",
+    },
+    {
+      label: "Settings",
+      icon: <PiGearSix className="text-2xl" />,
+      path: "account/security",
+    },
+    {
+      label: "Notifications",
+      icon: <FaRegBell className="text-2xl" />,
+      path: "/user/overview",
+      mobileOnly: true,
+    },
+  ];
+
```

Comment:
you can also move this to another file incase we want to add more content on the sidebar

Reply:
okay will do